src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4122,7 +4122,23 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
                     // can succeed.  This call will mount the drive if it wasn't already.
                     executionContext.SessionState.Drive.GetAtScope(wordToComplete.Substring(0, 1), "global");
                 }
+#if UNIX
+                StringBuilder sb = new StringBuilder();
+
+                foreach (var ch in wordToComplete.ToCharArray())
```

Comment:
You don't need to call `ToCharArray` - that will unnecessarily create an array.

In C#, strings are enumerable, so you would just use `foreach (var ch in wordToComplete) ...`.

Reply:
First I made a prototype on PowerShell. :-)
Fixed.